pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/UnnecessaryCaseChangeRule.java

```diff
@@ -30,16 +30,18 @@ public Object visit(ASTPrimaryExpression expr, Object data) {
             if (hasEqualsMethodCallChainedAtPosition(expr, chainedMethodCallIndex)
                     || isArgumentOfEqualsMethodCall(expr)) {
                 addViolation(data, expr);
+                return data;
```

Comment:
Since we don't use rule chain, you need to `return super.visit(expr, data)` here.... just returning data would skip nested classes etc.

Reply:
Ohh, sorry. I thought it would be okay to skip processing nested expressions in case we have already reported a problem here. I'll fix that